packages/signers/src/magic/__tests__/signer.test.ts

```diff
@@ -0,0 +1,21 @@
+import { Magic } from "magic-sdk";
+import { MagicSigner } from "../signer.js";
+
+describe("Magic Signer Tests", () => {
+  it("should correctly error when signing message if not authenticated", async () => {
+    const signer = await givenSigner();
+
+    const signMessage = signer.signMessage("test");
+    await expect(signMessage).rejects.toThrowErrorMatchingInlineSnapshot(
+      '"No signer found"'
+    );
+  });
+});
+
+const givenSigner = async () => {
+  const inner = new Magic("test");
```

Comment:
one way you can test this since this won't authenticate correctly is to:
1) test un-authed by just counting on `this.signer` not being set since you didn't call authenticate
2) test authed by having `params.authenticate` no-op and then you can mock out `this.inner.wallet.getProvider()` to return something like a `LocalWallet` or something that will give you consistent signing results